lib/ts/recipe/emailpassword/emailPasswordAuth.tsx

```diff
@@ -32,53 +32,50 @@ type Props = FeatureBaseProps & {
     onSessionExpired?: () => void;
 };
 
-class EmailPasswordAuth extends PureComponent<Props> {
-    /*
-     * Render.
-     */
-    render = (): JSX.Element | null => {
-        const emailVerification = (
-            <EmailVerificationAuth recipe={this.props.recipe.emailVerification} history={this.props.history}>
-                {this.props.children}
-            </EmailVerificationAuth>
-        );
+function EmailPasswordAuth(props: Props) {
+    const emailVerification = (
+        <EmailVerificationAuth recipe={props.recipe.emailVerification} history={props.history}>
+            {props.children}
+        </EmailVerificationAuth>
+    );
 
-        if (this.props.requireAuth === false) {
-            return <SessionAuth onSessionExpired={this.props.onSessionExpired}>{emailVerification}</SessionAuth>;
-        }
+    if (props.requireAuth === false) {
+        return <SessionAuth onSessionExpired={props.onSessionExpired}>{emailVerification}</SessionAuth>;
+    }
 
-        return (
-            <SessionAuth
-                redirectToLogin={() =>
-                    EmailPassword.getInstanceOrThrow().redirectToAuthWithRedirectToPath(undefined, this.props.history)
-                }
-                requireAuth={true}
-                onSessionExpired={this.props.onSessionExpired}>
-                {emailVerification}
-            </SessionAuth>
-        );
-    };
+    return (
+        <SessionAuth
+            redirectToLogin={() =>
+                EmailPassword.getInstanceOrThrow().redirectToAuthWithRedirectToPath(undefined, props.history)
+            }
+            requireAuth
+            onSessionExpired={props.onSessionExpired}>
+            {emailVerification}
+        </SessionAuth>
+    );
 }
 
+const EmailPasswordAuthMemo = memo(EmailPasswordAuth);
+
 export default function EmailPasswordAuthWrapper({
     children,
     requireAuth,
     onSessionExpired,
 }: {
-    children: JSX.Element;
+    children: React.ReactNode;
```

Comment:
To test this out, you should modify test/with-typescript/src/App.tsx to have a react component which uses this component and renders multiple children inside it. Then change the tsconfig in that project in such a way that it would throw an error if this was reverted back to `JSX.Element`.

Reply:
@xuatz perhaps you missed this comment?

Reply:
I did see this comment. I was going to setup my local for testing according to the instructions on CONTRIBUTING, because i just `npm run test` wasnt working somehow. Then it said i had to get the TOKENs from you guys on discord then i ran out of time for the day ^^

I think most of the changes are done bar validating with a test, so if you are not in a rush, I can follow it up later this week. Or you can finish up the PR if you dont mind 👍

Reply:
I'll finish the PR :) Thanks